logbook-core/src/main/java/org/zalando/logbook/core/FilteredHttpResponse.java

```diff
@@ -1,24 +1,32 @@
 package org.zalando.logbook.core;
 
-import lombok.AllArgsConstructor;
+import java.io.IOException;
 import org.apiguardian.api.API;
 import org.zalando.logbook.BodyFilter;
 import org.zalando.logbook.ForwardingHttpResponse;
 import org.zalando.logbook.HeaderFilter;
 import org.zalando.logbook.HttpHeaders;
 import org.zalando.logbook.HttpResponse;
-
-import java.io.IOException;
+import org.zalando.logbook.attributes.HttpAttributes;
 
 import static org.apiguardian.api.API.Status.INTERNAL;
 
 @API(status = INTERNAL)
-@AllArgsConstructor
 final class FilteredHttpResponse implements ForwardingHttpResponse {
 
     private final HttpResponse response;
     private final HeaderFilter headerFilter;
     private final BodyFilter bodyFilter;
+    private final HttpAttributes attributes;
+
+    FilteredHttpResponse(final HttpResponse response,
+                         final HeaderFilter headerFilter,
+                         final BodyFilter bodyFilter) {
+        this.response = response;
+        this.headerFilter = headerFilter;
+        this.bodyFilter = bodyFilter;
+        this.attributes = response.getAttributes();
```

Comment:
Could you add a bit more context about how/why does it not work right now? [ForwardingHttpResponse would also just get attributes from the original response](https://github.com/zalando/logbook/blob/f434a7e22993cf18f119dc87100051337f2e3e4b/logbook-api/src/main/java/org/zalando/logbook/ForwardingHttpResponse.java#L38), how does extracting the attributes earlier help?